### PR TITLE
Add json sphinx

### PR DESCRIPTION
**Pull request by wkerzendorf**
This will add our schemas to the sphinx documentation

### CODE REVIEW

**Comment by wkerzendorf** [comment]
@unoebauer I'd like to merge this.

File: docs/configuration/configuration.rst [hunk 1]
```diff
@@ -2,5 +2,168 @@
...
+-----------
+
+The ``montecarlo`` section describes the parameters for the MonteCarlo radiation transport and convergence criteria:
+
+.. jsonschema:: schemas/montecarlo.yml
+
+The ``seed`` parameter seeds the random number generator first for the creation of the packets
+(:math:`\nu` and :math:`\mu`) and then the interactions in the actual MonteCarlo process.
+The ``no_of_packets`` parameter can take a float number for input convenience and gives the number of packets normally
+used in each MonteCarlo loop. The parameters ``last_no_of_packets`` and ``no_of_virtual_packets`` influence the last run
```
Comment (unoebauer) [inline docs/configuration/configuration.rst]: specify that you refer to the MC cycles in which the plasma state is iteratively found

File: docs/configuration/configuration.rst [hunk 2]
```diff
@@ -2,5 +2,168 @@
...
+of the MonteCarlo loop when the radiation field should have converged. ``last_no_of_packets`` is normally higher than
```
Comment (unoebauer) [inline docs/configuration/configuration.rst]: specify that this is the run that calculates the final real and virtual spectra

File: docs/configuration/configuration.rst [hunk 3]
```diff
@@ -2,5 +2,168 @@
...
+``no_of_packets`` to create a less noisy output spectrum. ``no_of_virtual_packets`` can also be set to greater than 0 to
```
Comment (unoebauer) [inline docs/configuration/configuration.rst]: Make a note that this should typically be of the order of 1 and that increasing this number drastically increases the computational costs (and memory requirements if --with-vpacket-logging is active). So it should we used with care. According to SAS 3 was always a good compromise.

File: docs/configuration/configuration.rst [hunk 4]
```diff
@@ -2,5 +2,168 @@
...
+use the Virtual Packet formalism (reference missing ?????). The ``iterations`` parameter describes the maximum number of
+MonteCarlo loops executed in a simulation before it ends. Convergence criteria can be used to make the simulation stop
+sooner when the convergence threshold has been reached.
+
+The ``convergence_criteria`` section again has a ``type`` keyword. Two types are allowed: ``damped`` and ``specific``.
+All convergence criteria can be specified separately for the three variables for which convergence can be checked
```
Comment (unoebauer) [inline docs/configuration/configuration.rst]: Not anymore! There is only damped!

File: docs/configuration/configuration.rst [hunk 5]
```diff
@@ -2,5 +2,168 @@
...
+(``t_inner``, ``t_rad``, ``ws``) by specifying subsections in the ``convergence_criteria`` of the same name. These
+override then the defaults.
+
+
+#. ``damped`` only has one parameter ``damping-constant`` and does not check for convergence.
+
+#. ``specific`` checks for the convergence threshold specified in ``threshold``. For ``t_rad`` and ``w`` only a given
+    fraction (specified in ``fraction``) has to cross the ``threshold``. Once a convergence  threshold is read, the simulation
```
Comment (unoebauer) [inline docs/configuration/configuration.rst]: Deprecated

File: docs/configuration/configuration_old.rst [hunk 6]
```diff
@@ -1,8 +1,8 @@
 .. _config-file:
 
-******************
-Configuration File
-******************
+**********************
+Old Configuration File
+**********************
```
Comment (unoebauer) [inline docs/configuration/configuration_old.rst]: Do you want to keep this? I would be for just removing this

File: docs/configuration/configuration.rst [hunk 7]
```diff
@@ -2,5 +2,168 @@
 Configuration File
 ******************
 
+TARDIS uses the `YAML markup language <https://en.wikipedia.org/wiki/YAML>`_
+for its configuration files. There are several sections which allow different
+settings for the different aspects of the TARDIS calculation. An example
+configuration file can be downloaded :download:`here
+<../examples/tardis_example.yml>`.
+
```
[the remark this answers is not in the document]
Comment (wkerzendorf) [reply docs/configuration/configuration.rst]: done

**Comment by wkerzendorf** [comment]
@unoebauer it feels like a referee report ;-)

File: docs/configuration/configuration.rst (hunk 7 above)
[the remark this answers is not in the document]
Comment (wkerzendorf) [reply docs/configuration/configuration.rst]: done

File: docs/configuration/configuration.rst [hunk 8]
```diff
@@ -2,5 +2,168 @@
...
+In the ``structure`` section, one can specify a ``file`` section containing a ``type`` parameter
+(currently only ``artis`` is supported``) and a ``name`` parameter giving a path top a file. For the ``artis`` type,
+one can specify the inner and outermost shell by giving a ``v_lowest`` and ``v_highest`` parameter. This will result in
+the selection of certain shells which will be obeyed in the abundance section as well if ``artis`` is selected there as
+well.
+
+.. warning::
+    If a ``file`` section is given, all other parameters and sections in the ``structure`` section are ignored!
+
```
[the remark this answers is not in the document]
Comment (wkerzendorf) [reply docs/configuration/configuration.rst]: no it's not true and we have changed it.

File: docs/configuration/configuration.rst (hunk 4 above)
Comment (wkerzendorf) [reply docs/configuration/configuration.rst]: removed specific

File: docs/configuration/configuration.rst (hunk 5 above)
Comment (wkerzendorf) [reply docs/configuration/configuration.rst]: not sure what you mean - check the new version.

**Pull request merged**